src/Elliptic/Systems/Poisson/Equations.hpp

```diff
@@ -0,0 +1,168 @@
+    // the polynomial degree $p$ on both sides of the mortar, this should be
+    // changed to be just the parameter multiplying $\frac{p^2}{h}$.
+    static constexpr OptionString help = {
+        "The prefactor to the penalty term of the flux."};
+  };
+  using options = tmpl::list<PenaltyParameter>;
+  static constexpr OptionString help = {
+      "Computes the internal penalty flux for a poisson system."};
+
+  InternalPenaltyFlux() = default;
+  InternalPenaltyFlux(double penalty_parameter)
+      : penalty_parameter_(std::move(penalty_parameter)) {}
+
+  // clang-tidy: non-const reference
+  void pup(PUP::er& p) noexcept { p | penalty_parameter_; }  // NOLINT
+
+  struct NormalTimesFieldFlux : db::SimpleTag {
+    using type = tnsr::i<DataVector, Dim, Frame::Inertial>;
+    static std::string name() noexcept { return "NormalDotGradFieldFlux"; }
+    static constexpr bool should_be_sliced_to_boundary = true;
+  };
+
+  struct NormalDotGradFieldFlux : db::SimpleTag {
+    using type = Scalar<DataVector>;
+    static std::string name() noexcept { return "NormalDotGradFieldFlux"; }
+    static constexpr bool should_be_sliced_to_boundary = true;
+  };
+
+  // These tags are sliced to the interface of the element and passed to
+  // `package_data` to provide the data needed to compute the numerical fluxes.
+  using argument_tags =
+      tmpl::list<LinearSolver::Tags::Operand<Field>,
+                 Tags::deriv<LinearSolver::Tags::Operand<Field>,
+                             tmpl::size_t<Dim>, Frame::Inertial>,
+                 Tags::Normalized<Tags::UnnormalizedFaceNormal<Dim>>>;
+
+  // This is the data needed to compute the numerical flux.
+  // `SendBoundaryFluxes` calls `package_data` to store these tags in a
+  // Variables. Local and remote values of this data are then combined in the
+  // `()` operator.
+  using package_tags = tmpl::list<LinearSolver::Tags::Operand<Field>,
+                                  NormalTimesFieldFlux, NormalDotGradFieldFlux>;
+
+  // Following the packaged_data pointer, this function expects as arguments the
+  // types in `argument_tags`.
+  void package_data(gsl::not_null<Variables<package_tags>*> packaged_data,
+                    const Scalar<DataVector>& field,
+                    const tnsr::i<DataVector, Dim, Frame::Inertial>& grad_field,
+                    const tnsr::i<DataVector, Dim, Frame::Inertial>&
+                        interface_unit_normal) const noexcept;
+
+  // This function provides an opportunity to transform what is used as remote
+  // data on external boundaries. An empty implementation means that the local
+  // data is copied and used as remote data as-is. However, sign flips must be
+  // performed here for quantities that involve the face normal. Furthermore, to
+  // impose vanishing Dirichlet boundary conditions we can invert the remote
+  // field value here.
+  void transform_boundary_data(
```

Comment:
is this just a hack to impose boundary conditions?

Reply:
yes